calendar-bundle/tests/Security/Voter/CalendarContentVoterTest.php

```diff
@@ -0,0 +1,202 @@
+
+        $this->assertTrue($voter->supportsAttribute(ContaoCorePermissions::DC_PREFIX.'tl_content'));
+        $this->assertTrue($voter->supportsType(ReadAction::class));
+        $this->assertTrue($voter->supportsType(CreateAction::class));
+        $this->assertTrue($voter->supportsType(UpdateAction::class));
+        $this->assertTrue($voter->supportsType(DeleteAction::class));
+        $this->assertFalse($voter->supportsAttribute('foobar'));
+        $this->assertFalse($voter->supportsAttribute(ContaoCorePermissions::DC_PREFIX.'tl_page'));
+    }
+
+    /**
+     * @dataProvider checksElementAccessPermissionProvider
+     */
+    public function testChecksElementAccessPermission(CreateAction|DeleteAction|ReadAction|UpdateAction $action, array $parentRecords, array $events): void
+    {
+        $token = $this->createMock(TokenInterface::class);
+
+        $accessDecisionMap = [[$token, [ContaoCalendarPermissions::USER_CAN_ACCESS_MODULE], null, true]];
+
+        foreach ($events as $calendarId) {
+            $accessDecisionMap[] = [$token, [ContaoCalendarPermissions::USER_CAN_EDIT_CALENDAR], $calendarId, true];
+        }
+
+        $accessDecisionManager = $this->createMock(AccessDecisionManagerInterface::class);
+        $accessDecisionManager
+            ->expects($this->exactly(max(\count($parentRecords), \count($events)) * 2))
+            ->method('decide')
+            ->willReturnMap($accessDecisionMap)
+        ;
+
+        $fetchAllAssociativeMap = [];
+
+        foreach ($parentRecords as $id => $records) {
+            $fetchAllAssociativeMap[] = [
+                'SELECT id, @pid:=pid AS pid, ptable FROM tl_content WHERE id=?'.str_repeat(' UNION SELECT id, @pid:=pid AS pid, ptable FROM tl_content WHERE id=@pid', 9),
+                [$id],
+                [],
+                $records,
+            ];
+        }
+
+        $fetchOneMap = [];
+
+        foreach ($events as $id => $pid) {
+            $fetchOneMap[] = [
+                'SELECT pid FROM tl_calendar_events WHERE id=?',
+                [$id],
+                [],
+                $pid,
+            ];
+        }
+
+        $connection = $this->createMock(Connection::class);
+        $connection
+            ->expects($this->exactly(\count($parentRecords)))
+            ->method('fetchAllAssociative')
+            ->willReturnMap($fetchAllAssociativeMap)
+        ;
+
+        $connection
+            ->expects($this->exactly(\count($events)))
+            ->method('fetchOne')
+            ->willReturnMap($fetchOneMap)
+        ;
+
+        $voter = new CalendarContentVoter($accessDecisionManager, $connection);
+        $decision = $voter->vote($token, $action, [ContaoCorePermissions::DC_PREFIX.'tl_content']);
+
+        $this->assertSame(VoterInterface::ACCESS_ABSTAIN, $decision);
+    }
+
+    public static function checksElementAccessPermissionProvider(): iterable
+    {
+        yield 'Check access to news archive when creating element in news' => [
```

Comment:
```suggestion
        yield 'Check access to calendar when creating element in events' => [
```

Needs to be changed in the lines below, too.